src/ng/directive/ngRepeat.js

```diff
@@ -652,6 +650,10 @@ var ngRepeatDirective = ['$parse', '$animate', '$compile', function($parse, $ani
             }
           }
           lastBlockMap = nextBlockMap;
+
+          // Clear the value property from the hashFnLocals object to prevent a reference to the last value
+          // being leaked into this ngRepeatCompile function scope
+          hashFnLocals[valueIdentifier] = undefined;
```

Comment:
WDYT about clearing the value here? This way we don't have to touch the `trackByIdExpFn` method that gets invoked on each element and just add this one assignment at the end.